Px.Utils/Validation/SyntaxValidation/SyntaxValidation.cs

```diff
@@ -0,0 +1,291 @@
+        /// </summary>
+        /// <param name="keyValuePairs">The list of <see cref="KeyValuePairValidationEntry"/> objects to be converted.</param>
+        /// <param name="syntaxConf">The syntax configuration for the PX file.</param>
+        /// <returns>A list of <see cref="StructuredValidationEntry"/> objects built from the provided list of <see cref="KeyValuePairValidationEntry"/> objects.</returns>
+        public static List<StructuredValidationEntry> BuildStructuredEntries(List<KeyValuePairValidationEntry> keyValuePairs, PxFileSyntaxConf syntaxConf)
+        {
+            return keyValuePairs.Select(entry =>
+            {
+                ValidationEntryKey key = ParseValidationEntryKey(entry.KeyValueEntry.Key, syntaxConf);
+                return new StructuredValidationEntry(entry.Line, entry.Character, entry.File, key, entry.KeyValueEntry.Value);
+            }).ToList();
+        }
+
+        private static List<StringValidationEntry> BuildStringEntries(Stream stream, Encoding encoding, PxFileSyntaxConf syntaxConf, string filename, int bufferSize)
+        {
+            SyntaxValidation.LineCharacterState state = new() { Line = 0, Character = 0 };
+            stream.Seek(0, SeekOrigin.Begin);
+            using StreamReader reader = new(stream, encoding);
+            List<StringValidationEntry> stringEntries = [];
+            StringBuilder entryBuilder = new();
+            char[] buffer = new char[bufferSize];
+
+            while (reader.Read(buffer, 0, bufferSize) > 0)
+            {
+                ProcessBuffer(buffer, syntaxConf, state, filename, stringEntries, entryBuilder);
+                Array.Clear(buffer, 0, buffer.Length);
+            }
+            return stringEntries;
+        }
+
+        private static async Task<List<StringValidationEntry>> BuildStringEntriesAsync(Stream stream, Encoding encoding, PxFileSyntaxConf syntaxConf, string filename, int bufferSize)
+        {
+            LineCharacterState state = new() { Line = 0, Character = 0 };
+            stream.Seek(0, SeekOrigin.Begin);
+            using StreamReader reader = new(stream, encoding);
+            List<StringValidationEntry> stringEntries = [];
+            StringBuilder entryBuilder = new();
+            char[] buffer = new char[bufferSize];
+
+            while (await reader.ReadAsync(buffer, 0, bufferSize) > 0)
+            {
+                ProcessBuffer(buffer, syntaxConf, state, filename, stringEntries, entryBuilder);
+                Array.Clear(buffer, 0, buffer.Length);
+            }
+            return stringEntries;
+        }
+
+        private static void ProcessBuffer(char[] buffer, PxFileSyntaxConf syntaxConf, LineCharacterState state, string filename, List<StringValidationEntry> stringEntries, StringBuilder entryBuilder)
+        {
+            for (int i = 0; i < buffer.Length; i++)
```

Comment:
Doing nested function calls in the hot loop may have a significant performance impact. I suggest trying to merge these functions.